### PR TITLE
feat/comment-space

### PR DESCRIPTION
**Pull request by dmarticus**
my attempt at resolving the issue I filed: #118

### CODE REVIEW

File: app/Main.hs [hunk 1]
```diff
@@ -128,7 +128,7 @@ formatOne CabalDefaultExtensionsOpts {..} mode config mpath =
   where
     configPlus dynOpts = config {cfgDynOptions = cfgDynOptions config ++ dynOpts}
 
-----------------------------------------------------------------------------
+-- --------------------------------------------------------------------------
```
Comment (dmarticus) [inline app/Main.hs:131]: oh ooops, my change caused this!  Let me make an exception to `---` characters, lol.

File: fourmolu.cabal [hunk 2]
```diff
@@ -6,6 +6,7 @@ license-file:       LICENSE.md
 maintainer:
     Matt Parsons <parsonsmatt@gmail.com>
     George Thomas <georgefsthomas@gmail.com>
+
```
[the remark this answers is not in the document]
Comment (dmarticus) [reply fourmolu.cabal:9]: oh, that's my bad.  Definitely was an accident that I didn't catch when self-reviewing.  I'll remove it!

**Comment by georgefst** [comment]
> My only concern is, how does Ormolu handle this? My gut reaction is that Ormolu has the same behavior, in which case, this should be fixed in Ormolu and then ported to Fourmolu by the usual process.

In general I agree. And I think we should have some guidance (probably in the form of an issue template) about considering asking a question upstream first. In this case it isn't a bug as such and it's possible that Tweag wouldn't want to "fix" it. But there's little harm in asking.

**Review by georgefst — CHANGES_REQUESTED** [review]
Needs at least one test involving Haddock comments. I worry that `--|comment` -> `-- | comment` is an AST change.

**Comment by georgefst** [comment]
> cool, ormolu merged in my changes with this PR [tweag/ormolu#809](https://github.com/tweag/ormolu/pull/809)
> 
> @georgefst, should we just upstream those changes here, or do you want me to make a separate PR that puts the rule behind a flag?

Cool! In that case I'm happy to just merge Ormolu as-is, and wait til someone complains about wanting to switch the behaviour off.

This should hopefully be a very simple merge. We have some guidance here: https://github.com/fourmolu/fourmolu/blob/master/DEVELOPER.md#merging-upstream. But if you don't want to bother going through that, one of us will get round to it eventually, probably when the next version of Ormolu is released.

Are you happy to wait a while for an actual release? We've never previously published stuff from Ormolu before they have, and I'd be wary of doing so.

**Comment by georgefst** [comment]
Implemented upstream. Feel free to subscribe to #126.

**Comment by dmarticus** [comment]
Hi @georgefst, sorry for all the questions, but what does 
> subscribe to #126
mean in this context?  Are you saying that once y'all merge in Ormolu's latest changes to Fourmolu and release it, my feature that I merged to Ormolu will be released for Fourmolu as well?

**Pull request closed without merging**